internal/controller/amaltheasession_controller.go

```diff
@@ -120,6 +120,17 @@ func (r *AmaltheaSessionReconciler) Reconcile(ctx context.Context, req ctrl.Requ
 		}
 	}
 
+	if customResourceNeedsDeletion(amaltheasession) {
+		err = r.Client.Delete(ctx, amaltheasession)
+		log.Info("custom resource deleted")
+		return ctrl.Result{}, err
+	}
+
+	err = updateStatefulSetReplicas(ctx, r, amaltheasession)
```

Comment:
Then this has to be changed too.

```suggestion
	err = updateHibernationState(ctx, r, amaltheasession)
```